### DISCUSSION TITLE
Question about type conversion between parquet and arrow.

### DISCUSSION THREAD

**Discussion by RinChanNOWWW** [apache/arrow-rs#4731] [Q&A] [answered]
`parquet_to_arrow_schema` converts `Type::Int96` to `DataType::Timstamp(TimeUnit::Nano)`.

However, `arrow_to_parquet_schema` converts `DataType::Timstamp(TimeUnit::Nano)` to `Type::Int96`.

Why?

**Comment by tustvold** [comment, marked as answer]
Int96 is deprecated - https://github.com/apache/parquet-format/blob/master/Encodings.md#plain-plain--0

Timestamps should always be encoded as int64 now - https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#timestamp

Int96 is only still supported for compatibility with spark, which still has issues supporting nanosecond timestamps stored as int64
